sceptre/cli/helpers.py

```diff
@@ -30,14 +29,14 @@ def catch_exceptions(func):
     def decorated(*args, **kwargs):
         """
         Invokes ``func``, catches expected errors, prints the error message and
-        exits sceptre with a non-zero exit code.
+        re-raises.
         """
         try:
             return func(*args, **kwargs)
         except (SceptreException, BotoCoreError, ClientError, Boto3Error,
                 TemplateError) as error:
             write(error)
-            sys.exit(1)
+            raise
```

Comment:
So I know I'm late to the game here, and this PR is a little old, but I have a couple thoughts here:
First:
> The main issue with this change is that users may well (and probably) do rely on status exit codes on the CLI. Removing the status code and replacing it with a dump of the stack trace isn't going to be a good solution.
@ngfgrant, I _believe_ blowing up with an exception _will_ result in a nonzero status code. Are you concerned that the code might not be 1?

Second:
@alexharv074 , have you considered using `traceback.print_exc()` instead of `write(error)` here? This would give you the full traceback and we could still have the `sys.exit(1)`

Finally: 
We could always have DEBUG_ERRORS environment variable or something like that which would reraise rather than exit with just the message. Has that been considered?

Reply:
@jfalkenstein This seems like a good compromise to me.